### PR TITLE
No number multiplication flag

### PR DESCRIPTION
**Pull request by sultur**
Added optional flag (default is off) for Greynir which disables the joining and multiplication of numbers written in text (e.g. 'tvö hundruð þúsund' would be three separate tokens with the flag set to True).

Currently runs all former tests with the flag set to true (apart from `test_parse.py:test_amounts()`, which assumes flag is not set).
Added the `test_no_multiply_numbers()` test function. Some of the `check_terminal` calls are commented out due to weird declensions for some terminals. We probably want some more varied tests using the flag.

### CODE REVIEW

**Comment by vthorsteinsson** [comment]
Should we consider merging this @sultur ?

**Comment by sultur** [comment]
Sure, we can probably merge this as it is only activated by a flag which by default is off. However, there are still some kinks that need to be ironed out regarding setting the case, gender and such of a number when it is tokenized (although the words are correctly split into tokens).

**Comment by sultur** [comment]
Ready for merging!
We now also split "10 milljón krónur" into multiple tokens, which wasn't in the original pull request (I think this behaviour is more logical as the flag is called no multiply numbers). I uncommented the tests which were bugging me and decided to ignore the variants for undeclinable numbers (they shouldn't matter).

The small change to `ResultDict` making `sc` a `Required` key (doesn't affect functionality) also helps for the upcoming pull requests to Greynir.

File: test/test_no_multiply_numbers.py [hunk 1]
```diff
@@ -0,0 +1,409 @@
+# type: ignore
+"""
+
+    test_no_multiply_numbers.py
+
+    Tests for Greynir no_multiply_numbers flag functionality
+
+    Copyright(C) 2021 by Miðeind ehf.
```
Comment (vthorsteinsson) [inline test/test_no_multiply_numbers.py:8]: Modify copyright year to 2022, and the Original author can be skipped.

File: test/test_no_multiply_numbers.py [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+        variants=[],
+    )
+    check_terminal(
+        t[3],
+        text="hundruðum",
+        lemma="hundrað",
+        category="no",
+        variants=["ft", "þgf", "hk"],
+    )
+    # TODO: gets interpreted as 'pundur:kk'?
```
Comment (vthorsteinsson) [inline test/test_no_multiply_numbers.py:246]: We need an entry in the preferences.conf file for this, i.e. to always prefer the stem "pund" to the stem "pundur" for the word forms that are common to the two stems.

File: src/reynir/reducer.py [hunk 3]
```diff
@@ -109,7 +109,7 @@
 
 class ResultDict(TypedDict, total=False):
     # Score
-    sc: int
+    sc: Required[int]
```
[the remark this answers is not in the document]
Comment (sultur) [reply src/reynir/reducer.py:112]: Yessir, I am currently running PyPy 7.3.9 which corresponds to Python 3.7.13.

File: test/test_matcher.py [hunk 4]
```diff
@@ -44,17 +44,17 @@
 
 
 @pytest.fixture(scope="module")
-def g():
+def r():
```
[the remark this answers is not in the document]
Comment (sultur) [reply test/test_matcher.py:47]: Just changed it to be consistent with the other test files, they all call this function `r` (noticed it as I was importing tests into the new test file).

File: test/test_no_multiply_numbers.py (hunk 2 above)
Comment (sultur) [reply test/test_no_multiply_numbers.py]: Nice! This fixes it beautifully after I added the line to `Prefs.conf`.

File: test/test_no_multiply_numbers.py (hunk 1 above)
Comment (sultur) [reply test/test_no_multiply_numbers.py]: Fixed!

File: test/test_no_multiply_numbers.py (hunk 2 above)
Comment (sultur) [reply test/test_no_multiply_numbers.py]: Weird, this fix doesn't work for the pypy3.6 tests on github. All the other tests pass though.

Comment (sveinbjornt) [reply test/test_no_multiply_numbers.py]: I believe we agreed to drop Python 3.6 support some weeks back. Maybe you (@sultur) can update the docs in this repo to state that 3.7 is required, rm 3.6 tests in workflow and add pypy-3.7? Does that sound good to you, @vthorsteinsson?

Comment (sveinbjornt) [reply test/test_no_multiply_numbers.py]: This change should probably also be documented in the release notes next time we release a new version of GreynirPackage on GitHub/PyPI.

Comment (vthorsteinsson) [reply test/test_no_multiply_numbers.py]: Yes, Python 3.7 as a minimum requirement is already decided, so we can safely drop the 3.6 stuff and change the documentation to that effect.

Comment (sveinbjornt) [reply test/test_no_multiply_numbers.py]: The biggest change between 3.6 and 3.7 is that dicts are guaranteed to be ordered in the latter. Is this being relied on somewhere in the code?

Comment (vthorsteinsson) [reply test/test_no_multiply_numbers.py]: No, we should not have been relying on that, although our own servers run PyPy which has always guaranteed dicts to be in insertion order. But of course we can start relying on dict insertion order now, and the OrderedDicts we have in the code can be re-examined to determine whether they are still needed.

File: .github/workflows/python-package.yml [hunk 5]
```diff
@@ -29,10 +29,10 @@ jobs:
         python -m pip install --upgrade pip wheel setuptools
         python -m pip install git+https://github.com/mideind/Tokenizer#egg=tokenizer
         python -m pip install -e .
-    - name: Type check with mypy
```
[the remark this answers is not in the document]
Comment (sultur) [reply .github/workflows/python-package.yml:32]: I commented the mypy run out for now, as there were too many warnings for me to fix in this PR at the moment. I would like to fix mypy's complaints sometime in the future though.

**Pull request merged**